indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -175,6 +175,7 @@
 
   private final ConcurrentMap<String, ScheduledFuture> removedWorkerCleanups = new ConcurrentHashMap<>();
   private final ProvisioningStrategy<WorkerTaskRunner> provisioningStrategy;
+  private final ExecutorService monitorSyncHandler;
```

Comment:
This extra service must be explicitly shutdown in `stop()`/`close()`, see https://github.com/code-review-checklists/java-concurrency#explicit-shutdown. Applies to other classes in this PR as well

Reply:
thank you, fixed.